04-onion-routing.md

```diff
@@ -525,6 +527,93 @@ same length. It's particularly useful when adding dummy hops at the end of a
 blinded route, to prevent the sender from figuring out which node is the final
 recipient.
 
+
+### Onion Message Payload Format
+
+Onion messages have an onion with an alternate `hop_payload`
+format: a `bigsize` followed by a `onionmsg_payload`.  Note that there
+is no legacy format, thus a `bigsize` of 0 means no payload.
+
+An intermediate node expects an `encrypted_data_tlv` which it can
+decrypt using the `blinding` which it is handed along with the onion
+message.
+
+Field numbers 64 and above are reserved for payloads for the final
+hop.
+
+1. `tlv_stream`: `onionmsg_payload`
+2. types:
+    1. type: 2 (`reply_path`)
+    2. data:
+        * [`point`:`first_node_id`]
+        * [`point`:`blinding`]
+        * [`...*onionmsg_path`:`path`]
+    1. type: 4 (`encrypted_data_tlv`)
+    2. data:
+        * [`...*byte`:`encrypted_data_tlv`]
+
+1. subtype: `onionmsg_path`
+2. data:
+    * [`point`:`node_id`]
+    * [`u16`:`enclen`]
+    * [`enclen*byte`:`encrypted_recipient_data`]
+
+
+#### Requirements
+
+The writer:
+- For the non-final nodes' `onionmsg_payload`:
+  - MUST set `enctlv` to a valid `encrypted_data_tlv` stream containing `next_node_id`.
```

Comment:
`enctlv` has been renamed in the format above but not in the requirements section

Reply:
Oops, fixed!